leader-election-impl/src/main/java/com/palantir/paxos/PaxosAcceptorState.java

```diff
@@ -117,4 +118,38 @@ public PaxosProposalId getLastAcceptedId() {
     public PaxosValue getLastAcceptedValue() {
         return lastAcceptedValue;
     }
+
+    /**
+     * Standard equals implementation, note that {@link #version} is not persisted and as such is not used in the
+     * calculation.
+     */
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Can we figure out a way to introduce a special method for this instead of relying on #equals? There's no harm in that, and relying on equals feels wrong, especially since you're ignoring some fields + if you have equals that forces you to have hashCode, which again we don't need.